app/views/viewlets/inspector-header.js

```diff
@@ -78,13 +78,13 @@ YUI.add('inspector-header-view', function(Y) {
       pojoModel.icon = utils.getIconPath(
           pojoModel.charmUrl, false,
           viewContainerAttrs.charmstore, viewContainerAttrs.env);
+      var name = pojoModel.displayName;
       if (pojoModel.pending) {
         // Check if there is already a service using the default name to
         // trigger the name ux.
         // This regex simply removes the outer parentheses from the
-        // displayName that is set in the ghost-inspector.js updateGhostName
-        // method.  If the regex doesn't match, blow up.  It should match.
-        var name = pojoModel.displayName.match(/^\(([^)]*)\)$/)[1];
+        // displayName that might be set in the model getter.
+        name = name.replace(/^\(/, '').replace(/$\)/, '');
```

Comment:
I agree making it a method doesn't work but as a generic helper it can be
write once vs the duplicate code currently.
On May 20, 2015 9:39 AM, "Jeff Pihach" <notifications@github.com> wrote:

> In app/views/viewlets/inspector-header.js
> <https://github.com/juju/juju-gui/pull/739#discussion_r30701762>:
>
> >        if (pojoModel.pending) {
> >          // Check if there is already a service using the default name to
> >          // trigger the name ux.
> >          // This regex simply removes the outer parentheses from the
> > -        // displayName that is set in the ghost-inspector.js updateGhostName
> > -        // method.  If the regex doesn't match, blow up.  It should match.
> > -        var name = pojoModel.displayName.match(/^\(([^)]*)\)$/)[1];
> > +        // displayName that might be set in the model getter.
> > +        name = name.replace(/^\(/, '').replace(/$\)/, '');
>
> See above.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/juju/juju-gui/pull/739/files#r30701762>.
>